### PR TITLE
Tsify map settings panel

### PR DESCRIPTION
**Pull request by zoran995**
### What this PR does

Fixes #<insert issue number here if relevant>

Convert map settings panel to typescript and use styled components. 
Fixed a check for Leaflet viewer so image optimisation options don't show when the viewer is 2D

### Checklist

-   [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated relevant documentation in `doc/`.
-   [ ] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/ReactViews/Map/Panels/SettingPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+    viewer.viewerMode = ViewerMode.Cesium;
+    viewer.viewerOptions.useTerrain = viewerMode === "3d";
+  } else if (viewerMode === "2d") {
+    viewer.viewerMode = ViewerMode.Leaflet;
+  } else {
+    console.error(
+      `Trying to select ViewerMode ${viewerMode} that doesn't exist`
+    );
+  }
+}
```
Comment (na9da) [inline lib/ReactViews/Map/Panels/SettingPanel.tsx]: I think it is a good decision to extract this out but maybe we should move it to `ViewerMode.ts` and also make `Terria.ts` use this function from `Terria.setViewerMode`. Let met know what you think.

**Review by na9da — CHANGES_REQUESTED** [review]
Thanks @zoran995 for the PR. It looks great. I have just made a small recommendation.

File: lib/ReactViews/Map/Panels/SettingPanel.tsx (hunk 1 above)
Comment (zoran995) [reply lib/ReactViews/Map/Panels/SettingPanel.tsx]: good point
it was a plan to extract it as a separate function when `Terria.setViewerMode` get merged but completely forgot about it

**Comment by zoran995** [comment]
@na9da I have extracted `setViewerMode` as you suggested. I have also taken a new version of Checkbox from our implementation and added it here as it is more flexible to use it with child elements instead of label prop

**Review by na9da — APPROVED** [review]
Thanks @zoran995 !

**Pull request merged**